engine/src/test/java/org/hibernate/validator/test/predefinedscope/LocaleResolverTest.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.test.predefinedscope;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Locale.LanguageRange;
+
+import javax.validation.MessageInterpolator;
+import javax.validation.Validation;
+import javax.validation.ValidatorFactory;
+import javax.validation.metadata.ConstraintDescriptor;
+
+import org.hibernate.validator.PredefinedScopeHibernateValidator;
+import org.hibernate.validator.messageinterpolation.HibernateMessageInterpolator;
+import org.hibernate.validator.spi.messageinterpolation.LocaleResolver;
+import org.hibernate.validator.spi.messageinterpolation.LocaleResolverContext;
+import org.hibernate.validator.testutil.TestForIssue;
+import org.testng.annotations.Test;
+
+@TestForIssue(jiraKey = "HV-1749")
+public class LocaleResolverTest {
+
+	@Test
+	public void testLanguageRangeSupport() throws NoSuchMethodException, SecurityException {
+		ValidatorFactory validatorFactory = getValidatorFactoryWithInitializedLocales( Locale.FRANCE, new Locale( "es", "ES" ) );
+		HibernateMessageInterpolator messageInterpolator = (HibernateMessageInterpolator) validatorFactory.getMessageInterpolator();
+
+		StaticFieldLocaleResolver.acceptLanguage = "fr-FR,fr;q=0.9";
```

Comment:
Knowing you have this solution, is the `HibernateMessageInterpolator` interface still useful? I'm not quite understanding how that interface is supposed to be used, to be honest.